src/Symfony/Component/Notifier/Bridge/Mastodon/MastodonTransport.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Mastodon;
+
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\ChatMessage;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Quentin Dequippe <quentin@dequippe.tech>
+ *
+ * @see https://docs.joinmastodon.org
+ */
+final class MastodonTransport extends AbstractTransport
+{
+    public function __construct(#[\SensitiveParameter] private readonly string $accessToken, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('mastodon://%s', $this->getEndpoint());
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof ChatMessage;
+    }
+
+    /**
+     * @see https://docs.joinmastodon.org/methods/statuses/
+     */
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof ChatMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, ChatMessage::class, $message);
+        }
+
+        $options = $message->getOptions()?->toArray() ?? [];
+        $options['status'] = $message->getSubject();
+
+        $endpoint = sprintf('https://%s/api/v1/statuses', $this->getEndpoint());
+
+        $response = $this->client->request('POST', $endpoint, [
```

Comment:
I'd suggest doing the same as the twitter PR: add a public request method to allow hitting any HTTP endpoint while being authenticated. This metthod should accept only host-relative URLs to prevent sending the auth info to any host by mistake.

Reply:
Do you mean something like that

```
public function request(string $method, string $url, array $options): ResponseInterface
{
    $url = sprintf('https://%s/%s', $this->getEndpoint(), $url);

    $options['auth_bearer'] = $this->accessToken;

    return $this->client->request($method, $url, $options);
}
```

Reply:
:+1: (I'd just remove the `/` between endpoint and url)